tensorflow_serving/batching/batching_util.cc

```diff
@@ -0,0 +1,229 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_serving/batching/batching_util.h"
+
+#include <string>
+
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/framework/types.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/lib/core/errors.h"
+
+
+namespace tensorflow {
+namespace serving {
+
+// Padding for one dimension of a tensor.
```

Comment:
I don't understand what this represents. What do "pad before" and "pad after" mean? Can you elaborate in the comments please.

Also, could "first" and "second" be renamed to "pad_before" and "pad_after"?